heartbeat/beater/manager.go

```
@@ -234,28 +230,44 @@ func createWatchUpdater(monitor *Monitor) func(content []byte) {
 	}
 }
 
-func createJob(client publisher.Client, r monitors.Job, name, typ string) scheduler.TaskFunc {
-	return createJobTask(client, r, name, typ)
+func createJob(client publisher.Client, task monitorTask) scheduler.TaskFunc {
+	return task.prepareSchedulerJob(client, task.job)
 }
 
-func createJobTask(client publisher.Client, r monitors.TaskRunner, name, typ string) scheduler.TaskFunc {
+func (m *monitorTask) prepareSchedulerJob(client publisher.Client, r monitors.TaskRunner) scheduler.TaskFunc {
 	return func() []scheduler.TaskFunc {
-		event, next, err := r.Run()
+		event, next, err := m.job.Run()
 		if err != nil {
 			logp.Err("Job %v failed with: ", err)
 		}
 
+		name := m.config.Name
+		if name == "" {
+			name = m.config.Name
+		}
+
 		if event != nil {
 			event.DeepUpdate(common.MapStr{
 				"monitor": common.MapStr{
 					"name": name,
-					"type": typ,
+					"type": m.config.Type,
 				},
 			})
 
 			if _, exists := event["type"]; !exists {
 				event["type"] = defaultEventType
 			}
+			err := common.MergeFields(event, m.config.EventMetadata.Fields, m.config.EventMetadata.FieldsUnderRoot)
+			if err != nil {
+				logp.Err("Failed to merge fields: %v", err)
+				return nil
```

Well, you right, should we also ignore errors on while add tags?